### ISSUE TITLE
[BUG] pkgrepo.managed on RHEL no longer idempotent

### ISSUE THREAD

**Issue by bsoldhogg** [saltstack/salt#59745]
**Description**
When running any `pkgrepo.managed` state against a RHEL based OS, the configuration is changed every time, despite it being successful.

**Setup**
State:
```
{% set version = 'version' %}
test-repo:
  pkgrepo.managed:
    - humanname: "Our test Repo"
    - name: test_repo
    - baseurl: http://repo.test.com/redhat/{{ grains['osmajorrelease'] }}/{{ grains['osarch'] }}/{{ version[:-2] }}
    - enabled: '1'
    - gpgcheck: '0'
```

**Steps to Reproduce the behavior**
The test repo does not exist on the test machines. It is created by Salt and then the state is run again which causes it to be created again.
This has been tested on RHEL 7.7 and OEL 7.9 with the same behaviour seen on both. The `pkgrepo.managed` state works as expected on any Debian based system we have.

**Expected behavior**
The state should create the repo on the first pass and, on any subsequent runs, make no further changes as the repo is in state.

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
Salt Version:
           Salt: 3001.6
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: 3.2.3
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.8
         Jinja2: 2.10
        libgit2: 0.26.0
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: 3.4.7
         pygit2: 0.26.2
         Python: 3.6.9 (default, Oct  8 2020, 12:12:24)
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: 17.1.2
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.5
 
System Versions:
           dist: ubuntu 18.04 Bionic Beaver
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-134-generic
         system: Linux
        version: Ubuntu 18.04 Bionic Beaver
```

The minions are also running 3001.6.

```
Salt Version:
               Salt: 3001.6
     
    Dependency Versions:
               cffi: Not Installed
           cherrypy: Not Installed
           dateutil: Not Installed
          docker-py: Not Installed
              gitdb: Not Installed
          gitpython: Not Installed
             Jinja2: 2.11.1
            libgit2: Not Installed
           M2Crypto: 0.35.2
               Mako: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.6.2
       mysql-python: Not Installed
          pycparser: Not Installed
           pycrypto: Not Installed
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 3.6.8 (default, May 26 2020, 12:56:00)
       python-gnupg: Not Installed
             PyYAML: 3.13
              PyZMQ: 17.0.0
              smmap: Not Installed
            timelib: Not Installed
            Tornado: 4.5.3
                ZMQ: 4.1.4
     
    System Versions:
               dist: ol 7.9 
             locale: UTF-8
            machine: x86_64
            release: 5.4.17-2011.4.6.el7uek.x86_64
             system: Linux
            version: Oracle Linux Server 7.9
```
</details>

**Additional context**

**Comment by piterpunk** [comment]
@bsoldhogg this seems to be the problem fixed in commit 59c9983, the significant changes were in `get_repo` function, basically was to change the first parameter from `name` to `repo` and the use of `name` inside the function. This fix wasn't backported to `3001.x`, but is present from `3002` and upwards.

**Comment by danielrobbins** [comment]
Going to close this issue as it was addressed in 3002. @bsoldhogg if you encounter any improper behavior related to RHEL and repos in 3002, please report this issue as a new issue and we will investigate.
